deltachat-ffi/src/lib.rs

```diff
@@ -1111,69 +1111,65 @@ pub unsafe extern "C" fn dc_chatlist_get_context(
 // dc_chat_t
 
 #[no_mangle]
-pub type dc_chat_t<'a> = dc_chat::dc_chat_t<'a>;
+pub type dc_chat_t<'a> = dc_chat::Chat<'a>;
```

Comment:
curious: why is the `<'a>` needed here?

Reply:
Because the Chat structure contains the context, but it can not outlive the context.  So the context member of chat is defined as having the same lifetime as chat itself, lifetime "a".  And such lifetime contraints make it into the types.